meson.build

```diff
@@ -0,0 +1,1629 @@
+    #'video/out/vo_direct3d.c',             'direct3d',
+    #'video/out/vo_rpi.c',                  'rpi-mmal',
+
+
+    #'osdep/macosx_application.m',          'cocoa',
+    #'osdep/macosx_events.m',               'cocoa',
+    #'osdep/macosx_menubar.m',              'cocoa',
+    #'osdep/macosx_touchbar.m',             'macos-touchbar',
+    #'osdep/mpv.rc',                        'win32-executable',
+    #'osdep/path-macosx.m',                 'cocoa',
+
+if not get_option('cplayer').disabled()
+    datadir = get_option('datadir')
+    confdir = get_option('sysconfdir')
+
+    conf_files = ['etc/mpv.conf', 'etc/input.conf',
+                  'etc/mplayer-input.conf', 'etc/restore-old-bindings.conf']
+    install_data(conf_files, install_dir: join_paths(datadir, 'doc', 'mpv'))
+
+    zsh = find_program('zsh', required: false)
+    if zsh.found()
+        zsh_install_dir = join_paths(datadir, 'zsh', 'site-functions')
+        install_data('etc/_mpv.zsh', install_dir: zsh_install_dir, rename: '_mpv')
+    endif
+
+    bash = find_program('bash', required: false)
+    if bash.found()
+        bash_install_dir = join_paths(datadir, 'bash-completion', 'completions')
+        install_data('etc/mpv.bash-completion', install_dir: bash_install_dir, rename: 'mpv')
+    endif
+
+    install_data('etc/mpv.desktop', install_dir: join_paths(datadir, 'applications'))
+    install_data('etc/encoding-profiles.conf', install_dir: join_paths(confdir, 'mpv'))
+
+    foreach size: ['16x16', '32x32', '64x64', '128x128']
+        icon_dir = join_paths('datadir', 'icons', 'hicolor', size, 'apps')
+        install_data('etc/mpv-icon-8bit-' + size + '.png', install_dir: icon_dir, rename: 'mpv.png')
+    endforeach
+
+    hicolor_dir = join_paths(datadir, 'icons', 'hicolor')
+    install_data('etc/mpv-gradient.svg', install_dir: join_paths(hicolor_dir, 'scalable', 'apps'), rename: 'mpv.svg')
+    install_data('etc/mpv-symbolic.svg', install_dir: join_paths(hicolor_dir, 'symbolc', 'apps'))
+endif
+
+if get_option('libmpv').enabled()
+    client_h_path = meson.source_root() + '/libmpv/client.h'
+    major = '''grep "#define MPV_CLIENT_API_VERSION" ''' + client_h_path + ''' | sed -e "s/.*(\(.*\),.*/\1/"'''
+    minor = '''grep "#define MPV_CLIENT_API_VERSION" ''' + client_h_path + ''' | sed -e "s/.* \(.*\)).*/\1/"'''
+    major_ver = run_command('sh', '-c', major).stdout().strip()
+    minor_ver = run_command('sh', '-c', minor).stdout().strip()
+    version = major_ver + '.' + minor_ver + '.0'
+
+    library('mpv', sources, dependencies: dependencies, version: version, install: true)
+
+    includedir = get_option('includedir')
+    libdir = get_option('libdir')
+
+    headers = ['libmpv/client.h', 'libmpv/opengl_cb.h', 'libmpv/render.h',
+               'libmpv/render_gl.h', 'libmpv/stream_cb.h']
+    install_data(headers, install_dir: join_paths(includedir, 'mpv'))
+    install_data('libmpv/mpv.pc.in', install_dir: join_paths(libdir, 'pkgconfig'), rename: 'mpv.pc')
```

Comment:
You'll need to configure this properly... I suggest using pkgconfig.generate instead, it will automatically handle e.g. PRIV_LIBS for you by following the dependencies list provided as the kwarg to `library('mpv')`.